### DISCUSSION TITLE
How to import a local package in Pluto environment?

### DISCUSSION THREAD

**Discussion by JaydevSR** [fonsp/Pluto.jl#1353] [Questions about using Pluto] [answered]
I want to use a package that i am working on in the pluto environment but I can not import it because pluto fails to recognize the package. Due to this I have to activate the base environment every time I want to use the package. Is there a way around it?

**Comment by disberd** [comment, marked as answer]
Unfortunately it is currently not straightforward to use the Pluto PkgManager with packages that are local or not registered.
I also work on some local packages that need to share with a few colleagues so I set up a privaty registry hosted on github with LocalRegistry.jl and use to be able to use my non-registered package without losing the convenience of the built-in package manager.

If you are planning to use Pluto to debug interactivly your own local package, I think it is still more convenient to just switch to standard environment without the pluto package manager.

**Reply by lungben** [reply]
Or use a local package registry: https://github.com/GunnarFarneback/LocalRegistry.jl
When you add this registry to your local Julia and register your packages in it, it also works with the new Pluto Pkg functionality.

Alternatively, see here: https://github.com/fonsp/Pluto.jl/wiki/%F0%9F%8E%81-Package-management

**Reply by markNZed** [reply]
The answer addresses the underlying question of how to work with local packages, but it breaks some of the nice aspects of Pluto package management. Maybe there is a missing feature? If a package is under development in the user's root environment then can Pluto search here first , then in the public repository? This might be nice for playing with changes to a package inside Julia. If packages are not dev'd in the user's root environment then the user would not know about this feature - it seems backward compatible for the beginner and if it does create problems it would be for experience devs who will have more luck understanding this behavior.

**Comment by drdozer** [comment]
I have what I think is a relate issue. I define a module in one cell. I can then use that module if I use names scoped by it. But I can't import it into the top-level scope. If I then make a new cell with a different module defined, that new module can't see the first module at all as far as I can tell.
